### PR TITLE
Mask vocab padding token ids from CE loss

### PR DESCRIPTION
**Pull request by maanug-nv**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/bert/model/base.py [hunk 1]
```diff
@@ -318,6 +322,22 @@ def forward(
...
+            unpadded_vocab_size = self.tokenizer.vocab_size
+
+            get_vocab_range = VocabUtility.vocab_range_from_per_partition_vocab_size
+            padded_vocab_size = logits.size()[-1]
+            rank = parallel_state.get_tensor_model_parallel_rank()
+            world_size = parallel_state.get_tensor_model_parallel_world_size()
+            vocab_start_index, _ = get_vocab_range(padded_vocab_size, rank, world_size)  # gets range on this tp rank
+
+            # mask tokens past unpadded_vocab_size. must be offset by where each tp rank's vocab range starts
+            mask_start = max(unpadded_vocab_size - vocab_start_index, 0)
```
Comment (sichu2023) [inline nemo/collections/llm/bert/model/base.py:338]: Nice.

**Review by sichu2023 — APPROVED** [review]
LGTM.
